src/BuildConfig.cc

```diff
@@ -842,16 +895,31 @@ BuildConfig::configureBuild() {
   const std::unordered_set<std::string> buildObjTargets =
       processSources(sourceFilePaths);
 
-  // Project binary target.
-  const std::string mainObjTarget = buildOutPath / "main.o";
-  std::unordered_set<std::string> projTargetDeps = { mainObjTarget };
-  collectBinDepObjs(
-      projTargetDeps, "",
-      targets.at(mainObjTarget).remDeps,  // we don't need sourceFile
-      buildObjTargets
-  );
+  if (executable) {
+    // Project binary target.
+    const std::string mainObjTarget = buildOutPath / "main.o";
+    std::unordered_set<std::string> projTargetDeps = { mainObjTarget };
+    collectBinDepObjs(
+        projTargetDeps, "",
+        targets.at(mainObjTarget).remDeps,  // we don't need sourceFile
+        buildObjTargets
+    );
+
+    defineLinkTarget(outBasePath / packageName, projTargetDeps);
+  }
 
-  defineLinkTarget(outBasePath / packageName, projTargetDeps);
+  if (library) {
+    // Project library target.
+    const std::string libTarget = buildOutPath / "lib.o";
+    std::unordered_set<std::string> libTargetDeps = { libTarget };
+    collectBinDepObjs(
+        libTargetDeps, "",
+        targets.at(libTarget).remDeps, // we don't need sourceFile
+        buildObjTargets
+    );
+
+    defineLibTarget(outBasePath / libName, libTargetDeps);
+  }
```

Comment:
These are almost equivalent. Can you make a new method for this?

Reply:
I made new methods but it leaves a few loose ends lying around now. The commands have to be duplicated since I couldn't think of a convenient way to do that. I want to make the command strings both static variables but the library command can't be static since it uses `fmt`. It might be able to be made static if we use a `make` variable instead of `fmt` but I'm not sure how we'd want to go about that. Thoughts?

Reply:
I was able to solve this a lot easier than I thought!